app/design/frontend/Magento/elogic-theme/Magento_Catalog/web/css/source/module/_listings.less

```diff
@@ -0,0 +1,171 @@
+
+    .product-item-link {
+        .lib-css(color, @color-elogic-gray-main);
+        .lib-css(font-weight, @font-weight-regular);
+        letter-spacing: 0.071rem;
+        line-height: 100%;
+    }
+
+    .product-item {
+        a:link,
+        a:visited {
+            .lib-css(color, @color-elogic-gray-main);
+        }
+    }
+
+    .product-item {
+        a:hover,
+        a:active {
+        .lib-css(color, @color-elogic-black);
+        text-decoration:none;
+        }
+    }
+
+
+    .product-image-container {
+        width: 100%;
+    }
+
+    .special-price {
+        .lib-css(font-weight, @font-weight-bold);
+        .lib-css(color, @color-elogic-black);
+        .lib-css(font-size, @font-size__l_xl);
+        line-height: 150%;
+    }
+    
+    .old-price {
+        display: inline;
+        text-decoration: line-through;
+        vertical-align: super;
+    }
+}
+
+//
+//  Mobile
+//  _____________________________________________
+
+.media-width(@extremum, @break) when (@extremum ='max') and (@break = @screen__m) {
+    .page-layout-1column .block.widget .products-grid .product-item {
+        margin-left: 0;
+        width: calc((100% - 4%) / 3);
+    }
+}
+
+.media-width(@extremum, @break) when (@extremum ='max') and (@break = @screen__s) {
+    .page-layout-1column .block.widget .products-grid .product-item {
+        margin-left: 0;
+        width: calc((100% - 4%) / 2);
+    }
+}
+
+.media-width(@extremum, @break) when (@extremum = 'max') and (@break = @screen__xs) {
+    .page-layout-1column .block.widget .products-grid .product-item {
+        width: 90%;
+        margin: 0 auto;
+    }
+}
+
+//
+//  Desktop
+//  _____________________________________________
+
+.media-width(@extremum, @break) when (@extremum = 'min') and (@break = @screen__m) {
+    .page-layout-1column .block.widget .block-content .products-grid .product-item {
+        width: calc((100% - 6%) / 3);
+    }
+
+    .page-layout-1column .block.widget .block-content .products-grid .product-item:nth-child(3n+1) {
+        margin-left: 0;
+    }
+}
+
+.media-width(@extremum, @break) when (@extremum = 'min') and (@break = @screen__l) {
+    .page-layout-1column .block.widget .block-content .products-grid .product-item {
+        width: calc((100% - 6%) / 4);
+    }
+
+    .page-layout-1column .block.widget .block-content .products-grid .product-item:nth-child(3n+1) {
+        margin-left: 0;
+    }
+
+    .page-layout-1column .block.widget .block-content .products-grid .product-item {
+        margin-left: 0;
+    }
+}
+
+.media-width(@extremum, @break) when (@extremum = 'min') and (@break = @screen__xl) {
+    .page-layout-1column .block.widget .block-content .products-grid .product-item {
+        width: calc((100% - 6%) / 4);
+    }
+}
+
+.media-width(@extremum, @break) when (@extremum = 'min') and (@break = @screen__lg) {
+    .page-layout-1column .block.widget .block-content .products-grid .product-item {
+        width: calc((100% - 6%) / 4);
+    }
+}
```

Comment:
you can use nesting for `.product-item` class

Reply:
Sorry, but do you mean visually nested code or what?
These classes are default magento classes for products widths + 1 class that i've added(.block-content) to add more specificity.